mathesar_ui/src/systems/users-and-permissions/UserDetailsForm.svelte

```diff
@@ -34,16 +34,13 @@
   $: isNewUser = user === undefined;
   $: fullName = optionalField(user?.full_name ?? '');
   $: username = requiredField(user?.username ?? '', [
-    validateLength(
-      150,
-      'Username should be lesser than or equal to 150 characters.',
-    ),
-    matchRegexAllowNullUndefined(
+    maxLength(150, 'Username cannot be longer than 150 characters.'),
+    matchRegex(
       /^[A-Za-z0-9_@.+-]*$/,
       'Username can only contain alphanumeric characters, _, @, +, ., and -.',
     ),
   ]);
-  $: email = optionalField(user?.email ?? '', [validateEmailAllowEmpty()]);
+  $: email = optionalField(user?.email ?? '', [isEmail()]);
```

Comment:
@seancolsen The case here is that email should allow empty values.

Try opening an user with an existing value for the email field, and clear it. This will result in a validation error.

Reply:
Thanks for pointing this out!

I added 24a2a67af4e86bb15dfeb47f25705b1ab449eb9a to fix this at the form level. Now the validation functions for `requiredField` _and_ `optionalField` use **filled** values. The validation logic returns early any time the value is considered empty. For required fields, empty values give a validation error. For optional fields, empty values bypass all subsequent validation.